crates/llms/src/nql/mod.rs

```diff
@@ -0,0 +1,116 @@
+/*
+Copyright 2024 The Spice.ai OSS Authors
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+     https://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+#![allow(clippy::missing_errors_doc)]
+use async_trait::async_trait;
+use snafu::Snafu;
+
+#[cfg(feature = "candle")]
+pub mod candle;
+
+#[cfg(feature = "mistralrs")]
+pub mod mistral;
+
+pub enum LlmRuntime {
+    Mistral,
+    Candle,
+}
+
+#[derive(Debug, Snafu)]
+pub enum Error {
+    #[snafu(display("Failed to run the NSQL model"))]
```

Comment:
For these generic errors, I would still leave in the source error message - otherwise its hard to debug what went wrong.

Reply:
I will look. Some of these don't have an error to propagate (e.g. when a file isn't found).

Reply:
No such errors